scripts/volmount/crypt/crypt

```diff
@@ -169,18 +175,52 @@ do_mount_disk()
 		echo "Unknown device mapper target"
 		exit 1
 	fi
-	[ $? -ne 0 ] && echo "ERROR: $dm_type: Unable to create/open crypt device" && exit 1
+	if [ $? -ne 0 ]; then
+		echo "ERROR: $dm_type: Unable to create/open crypt device"
+		exit 1
+	fi
 
 	if [ ! -b /dev/mapper/$device_name ]; then
-		echo "ERROR: could not crypt mapper"
-		return 2;
+		echo "ERROR: could not find the matching crypt mapper device /dev/mapper/$device_name"
+		return 2
 	fi
 
 	mkdir -p $3
-	if ! mount /dev/mapper/$device_name $3; then
-		echo "ERROR: could not setup dm-crypt volume"
-		return 3;
+	if mount /dev/mapper/$device_name $3; then
+		# only backup if a clean run success
+		[ -z "$specialrun" ] && cp -f $dir/$img_file.prebak $dir/$img_file.goodbak
+		sync
+		return 0
+	fi
+
+	echo "ERROR: could not setup dm-crypt volume"
+	if [ "$specialrun" == "fsck" ] && [ -f $dir/$img_file.goodbak ]; then
+		echo "Specialrun $specialrun did not succeed ... trying from goodbackup"
+		cp -f $dir/$img_file.goodbak $dir/$img_file
+		/usr/sbin/cryptsetup close $device_name || true
+		if ! do_mount_disk "$dm_type" "$path" "$mntpath" bak; then
```

Comment:
Will it be endless recursion here?

Reply:
it passes the "bak" argument at last position which should be the end run ... maybe check. i triggered that case manually as  i couldnt get fsck to not produce something mountable (but there was data loss which is not taht great).